dataprep/clean/clean_url.py

```diff
@@ -0,0 +1,229 @@
+# pylint: disable=too-many-arguments, trailing-whitespace, global-statement
+"""
+implement clean_url functionality
+"""
+
+import re
+from typing import Any, Union, Dict, List
+from urllib.parse import unquote, urlparse
+
+import pandas as pd
+import dask
+import dask.dataframe as dd
+
+from .utils import NULL_VALUES, to_dask
+
+# to extract queries
+QUERY_REGEX = r"(\?|\&)([^=]+)\=([^&]+)"
+
+# regex to validate url
+VALID_URL_REGEX = r"^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$"
+
+# removing auth params
+AUTH_VALUES = {"access_token", "auth_key", "auth", "password", "username", "login", "token"}
+
+# unified_list
+UNIFIED_AUTH_LIST = set()
+
+# STATs count
+STATS = {"cleaned": 0, "rows": 0, "correct_format": 0, "incorrect_format": 0}
+
+
+def clean_url(
+    df: Union[pd.DataFrame, dd.DataFrame],
+    column: str,
+    inplace: bool = False,
+    split: bool = False,
+    remove_auth: Union[bool, List[str]] = False,
+    report: bool = True,
```

Comment:
Add `error` parameter with value field `{"coerce","raise","ignore"}`. (Refer to current `clean_x()` functions)